R/import_instruments.R

```diff
@@ -21,136 +19,140 @@
 #' @export
 #'
 #' @examples
-#'\dontrun{
-#'  import_instruments(
-#'     "https://redcap.miami.edu/api/", 
-#'     Sys.getenv("test_API_key")
-#'  )
-#'}   
-import_instruments <- function(url, token, drop_blank = TRUE, envir = .GlobalEnv) {
-  #browser()
-  
-  ds_instrument <- 
-    REDCapR::redcap_metadata_read(redcap_uri=url, token=token)$data
-  
+#' \dontrun{
+#' import_instruments(
+#'   "https://redcap.miami.edu/api/",
+#'   Sys.getenv("test_API_key")
+#' )
+#' }
+import_instruments <- function(url, token, drop_blank = TRUE,
+                               record_id="record_id",
+                               envir = .GlobalEnv) {
+  # browser()
+  ds_instrument <-
+    REDCapR::redcap_metadata_read(redcap_uri = url, token = token)$data
+
   # Get names of instruments
   form_name <- NULL
-  
-  instrument_name <- ds_instrument |> 
-    pull(form_name) |> 
-    unique() 
 
-  
+  instrument_name <- ds_instrument |>
+    pull(form_name) |>
+    unique()
+
+
   # do the api call
-  #redcap <- redcapAPI::exportRecords(connection)
-  
-  raw_labels <- 
+  # redcap <- redcapAPI::exportRecords(connection)
+
+  raw_labels <-
     REDCapR::redcap_read(
-      redcap_uri = url, 
+      redcap_uri = url,
       token = token,
       raw_or_label_headers = "label",
       records = 1
     )$data
-  
+
   just_labels <- raw_labels
 
-  raw_redcapr <- 
-    REDCapR::redcap_read(
-      redcap_uri = url, 
+  raw_redcapr <-
+    REDCapR::redcap_read_oneshot(
+      redcap_uri = url,
       token = token,
     )$data
-  
+
   just_data <- raw_redcapr
-  
-  just_data[] <- 
-    mapply(nm = names(just_data),
-           lab = relabel(names(just_labels)),
-           FUN = function(nm, lab) {
-             labelVector::set_label(just_data[[nm]], lab)
-           }, 
-           SIMPLIFY = FALSE
+
+  just_data[] <-
+    mapply(
+      nm = names(just_data),
+      lab = relabel(names(just_labels)),
+      FUN = function(nm, lab) {
+        labelVector::set_label(just_data[[nm]], lab)
```

Comment:
add importfrom labelVector in Roxygen2 header add dependency to the pakcage for labelVector